src/pool/srv_pool.c

```diff
@@ -5912,6 +5967,49 @@ pool_upgrade_props(struct rdb_tx *tx, struct pool_svc *svc, uuid_t pool_uuid, cr
 		need_commit = true;
 	}
 
+	/*
+	 * Initialize server pool and container handles in the DB. To be conservative, we require
+	 * the old server pool and container handles to be initialized already in memory, and use
+	 * their existing values instead of generating new UUIDs.
+	 */
+	d_iov_set(&value, valuuid, sizeof(uuid_t));
+	rc = rdb_tx_lookup(tx, &svc->ps_root, &ds_pool_prop_srv_handle, &value);
+	if (rc && rc != -DER_NONEXIST) {
+		D_GOTO(out_free, rc);
+	} else if (rc == -DER_NONEXIST) {
+		if (uuid_is_null(svc->ps_pool->sp_srv_pool_hdl)) {
+			D_ERROR(DF_UUID ": server pool handle unstable\n", DP_UUID(pool_uuid));
+			rc = -DER_AGAIN;
```

Comment:
[Note] My manual testing has revealed a problem in this case. I think it's an existing bug with the upgrade mechanism, and will address it elsewhere.